### DISCUSSION TITLE
Can I have component inside flat-files convention folder?

### DISCUSSION THREAD

**Discussion by cleytoncarvalho** [kiliman/remix-flat-routes#84] [Q&A] [answered]
I have the following route structure:

-> course.$couderId.modules+
--> index.tsx
--> new.tsx

Can I have a component that is not matched as a route or layout inside this folder? (course.$couderId.modules+)

**Comment by cleytoncarvalho** [comment, marked as answer]
I did the following and it worked (I needed some collocated components with the index.tsx route):

-> course.$couderId.modules+
--> index
---> _index.tsx
---> component1.tsx
---> component2.tsx
--> new.tsx

**Reply by kiliman** [reply]
Yes, hybrid routes support colocation by creating a folder without the `+` suffix to store your other files. Your route module can be named `index|route|layout|page` or anything starting with `_`.
